File: frontend/src/types/product.ts

export interface KLSProduct {
    code: string;
    viet: string;
    eng: string;
    alternative: string | null;
    brand: string;
}

export interface AesculapProduct {
    code: string;
    viet: string;
    eng: string;
    image: string;
    alternative: string | null;
    brand: string;
}

export interface KLSImageResponse {
    code: string;
    img1_url: string;
    img2_url: string;
    img3_url: string;
}

File: server/app/schemas/product.py
from pydantic import BaseModel, ConfigDict


class ProductBase(BaseModel):
    code: str
    eng: str
    viet: str
    brand: str
    
    model_config = ConfigDict(from_attributes=True) 


class KLSResponse(ProductBase):
    alternative: str | None = None
    pass

class AesculapResponse(ProductBase):
    image: str | None = None
    alternative: str | None = None

class KLSImageResponse(BaseModel):
    code: str
    img1_url: str | None = None
    img2_url: str | None = None
    img3_url: str | None = None
    
    model_config = ConfigDict(from_attributes=True)


class KLSReportResponse(BaseModel):
    no: int
    code: str
    description: str
    img_url: str | None = None
    quantity: int
    
    model_config = ConfigDict(from_attributes=True)